### ISSUE TITLE
demo.py or test.py

### ISSUE THREAD

**Issue by ykzou9** [cnexah/VA-DepthNet#9]
Thank you very much for your sharing. I am very interested in your research and would like to politely ask if you have a demo.py or test.py file that I can use to test a single image with the trained model and obtain a depth map.

**Comment by cnexah** [comment]
Sorry for the late reply.
Could you please try the following code? (if the image size is 480x640)

import torch
from PIL import Image
import numpy as np
from vadepthnet.networks.vadepthnet import VADepthNet
from vadepthnet.dataloaders.dataloader import ToTensor

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
print("device: %s" % device)

model = VADepthNet(max_depth=10,
prior_mean=1.54,
img_size=(480, 640))
model = torch.nn.DataParallel(model).cuda()
checkpoint = torch.load('vadepthnet_nyu.pth', map_location=device)
model.load_state_dict(checkpoint['model'])
model.eval()
img = Image.open(image_path)
img = np.asarray(img, dtype=np.float32) / 255.0
#img = torch.from_numpy(img).cuda().unsqueeze(0)

totensor = ToTensor('test')
img = totensor.to_tensor(img)
img = totensor.normalize(img)
img = img.unsqueeze(0).cuda()
pdepth = model.forward(img)
print(pdepth)

**Comment by ykzou9** [comment]
> Sorry for the late reply. Could you please try the following code? (if the image size is 480x640)
> 
> import torch from PIL import Image import numpy as np from vadepthnet.networks.vadepthnet import VADepthNet from vadepthnet.dataloaders.dataloader import ToTensor
> 
> device = torch.device("cuda" if torch.cuda.is_available() else "cpu") print("device: %s" % device)
> 
> model = VADepthNet(max_depth=10, prior_mean=1.54, img_size=(480, 640)) model = torch.nn.DataParallel(model).cuda() checkpoint = torch.load('vadepthnet_nyu.pth', map_location=device) model.load_state_dict(checkpoint['model']) model.eval() img = Image.open(image_path) img = np.asarray(img, dtype=np.float32) / 255.0 #img = torch.from_numpy(img).cuda().unsqueeze(0)
> 
> totensor = ToTensor('test') img = totensor.to_tensor(img) img = totensor.normalize(img) img = img.unsqueeze(0).cuda() pdepth = model.forward(img) print(pdepth)

Thank you so much for taking the time to get back to me amidst your busy schedule. Your response has been immensely helpful, and I truly appreciate it. Wishing you smooth progress in your research endeavors.
